File: Crio/crio_programming_interview_problems-master/LetterCombinationsOfPhoneNumber/LetterCombinationsOfPhoneNumber.py
def letterCombinationsOfPhoneNumber(digits):
    # CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    res = []
    digits = str(digits)
    h = {
        '2':['a','b','c'],
        '3':['d','e','f'],
        '4':['g','h','i'],
        '5':['j','k','l'],
        '6':['m','n','o'],
        '7':['p','q','r','s'],
        '8':['t','u','v'],
        '9':['w','x','y','z']
    }
    for ch in h[digits[0]]:
        res.append(ch)
    for d in digits[1:]:
        t = []
        for r in res:
            for ch in h[d]:
                t.append(r+ch)
        res = t
    sorted(res)
    return res
    # CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

if __name__ == '__main__':
    digits = input()
    result = letterCombinationsOfPhoneNumber(digits)
    print(*result)

File: Crio/crio_programming_interview_problems-master/LetterCombinationsOfPhoneNumber/LetterCombinationsOfPhoneNumber.cpp
#include "../crio/cpp/io/FastIO.hpp"
#include "../crio/cpp/io/PrintMatrix.hpp"
#include <bits/stdc++.h>
using namespace std;

class LetterCombinationsOfPhoneNumber {
  public:
    // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    void rec(int index, string digits, vector<vector<char>> v,
             vector<string>& res, string str) {
        if (index >= int(digits.size())) {
            res.push_back(str);
            return;
        }
        for (int i = 0; i < int(v[digits[index] - '0'].size()); i++) {
            str[index] = v[digits[index] - '0'][i];
            rec(index + 1, digits, v, res, str);
        }
    }
    // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

    vector<string> letterCombinationsOfPhoneNumber(string digits) {
        vector<string> res;
        // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
        vector<vector<char>> v{{},
                               {},
                               {'a', 'b', 'c'},
                               {'d', 'e', 'f'},
                               {'g', 'h', 'i'},
                               {'j', 'k', 'l'},
                               {'m', 'n', 'o'},
                               {'p', 'q', 'r', 's'},
                               {'t', 'u', 'v'},
                               {'w', 'x', 'y', 'z'}};

        if (digits == "")
            return res;
        string str = digits;
        rec(0, digits, v, res, str);
        // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
        return res;
    }
};

int main() {
    FastIO();
    string digits;
    cin >> digits;
    vector<string> combinations =
        LetterCombinationsOfPhoneNumber().letterCombinationsOfPhoneNumber(digits);
    PrintMatrix<string>().OneDMatrix(combinations, " ");
    return 0;
}